larky/src/main/java/com/verygood/security/larky/parser/LarkyEvaluator.java

```diff
@@ -62,14 +66,33 @@ public LarkyEvaluator(LarkyScript larkyScript, Console console) {
     this.environment = createEnvironment(larkyScript.getBuiltinModules(), larkyScript.getGlobals());
   }
 
-  public Module eval(StarFile content)
+
+  @Builder
+  @Data
+  static class EvaluationResult {
```

Comment:
Please, consider defining a `public interface` like:

```
public interface EvaluationResult {
    boolean hasOutput();
    boolean hasModule();
    
    Object getOutput();
    Module getModule();
}

@Builder
@Data
class InternalEvaluationResult implements EvaluationResult {
    private Object output;
    private Module module;

    public boolean hasOutput() {
      return output != null;
    }

    public boolean hasModule() {
      return module != null;
    }
}

public EvaluationResult eval(StarFile content) {
...
}
```

This way we will have an immutable by default interface, and give a client of the API freedom to implement more specific implementation in their code(like if they want to check and hard type output, or do some checks, etc.)

Reply:
Good suggestion.

Reply:
Addressed https://github.com/verygoodsecurity/starlarky/pull/257/commits/0bb92f5a62ba7cc73ff96a94e3a8e127353e208e